### PR TITLE
Feature/deploy rabbit

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This PR includes the following proposed change(s):

- deploy rabbit to openshift-single instance

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested in DEV

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gitops/components/messaging/rabbitmq/cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Cluster RabbitMQ deployment on openshift
+
+As we cannot create role and role binding now, we cannot do RabbitMQ clustering.
```
Comment (pbolduc) [inline gitops/components/messaging/rabbitmq/cluster/README.md:3]: Chris and Adam can create role bindings, just not the rest of the team.

File: gitops/components/messaging/rabbitmq/single-instance/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+oc create -f secrets.yaml --save-config
+```
+
+5. create secrets for docker access (dockerhub credentials for pull in dockerhub images)
+
+```
+oc create secret docker-registry docker-creds ^
+--docker-server=docker.io ^
+--docker-username=bcgovfams3 ^
+--docker-password="@&W9H0e#eLlHv5&Z@pR1t" ^
```
Comment (pbolduc) [inline gitops/components/messaging/rabbitmq/single-instance/README.md:43]: Please redact and change the password changed ASAP

File: gitops/components/messaging/rabbitmq/single-instance/secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+kind: Secret
+apiVersion: v1
+metadata:
+  name: rabbitmq-secret
+data:
+  cookie: S1Y4MnhjNnA3MXJpU0ppZw==
```
Comment (pbolduc) [inline gitops/components/messaging/rabbitmq/single-instance/secrets.yaml:6]: We can't commit real secrets, be sure these are faked and call that out in the readme. shortly in the future we should be able to use Hashicorp Vault for our secrets

**Review by pbolduc — CHANGES_REQUESTED** [review]
looks good except for the credentials and secrets

File: gitops/components/messaging/rabbitmq/single-instance/secrets.yaml (hunk 3 above)
Comment (peggy-gov-bc-ca) [reply gitops/components/messaging/rabbitmq/single-instance/secrets.yaml:6]: sure

File: gitops/components/messaging/rabbitmq/single-instance/README.md (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply gitops/components/messaging/rabbitmq/single-instance/README.md]: changed

File: gitops/components/messaging/rabbitmq/cluster/README.md (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply gitops/components/messaging/rabbitmq/cluster/README.md]: change the wording

**Review by pbolduc — APPROVED** [review]

**Pull request merged**